pkg/cmd/variable/create/create.go

```diff
@@ -0,0 +1,223 @@
+				}
+
+				if opts.Visibility == shared.Selected && len(opts.RepositoryNames) == 0 {
+					return cmdutil.FlagErrorf("`--repos` list required with `--visibility=selected`")
+				}
+			} else {
+				if len(opts.RepositoryNames) > 0 {
+					opts.Visibility = shared.Selected
+				}
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return createRun(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.OrgName, "org", "o", "", "Create `organization` variable")
+	cmd.Flags().StringVarP(&opts.EnvName, "env", "e", "", "Create deployment `environment` variable")
+	cmdutil.StringEnumFlag(cmd, &opts.Visibility, "visibility", "v", shared.Private, []string{shared.All, shared.Private, shared.Selected}, "Create visibility for an organization variable")
+	cmd.Flags().StringSliceVarP(&opts.RepositoryNames, "repos", "r", []string{}, "List of `repositories` that can access an organization variable")
+	cmd.Flags().StringVarP(&opts.Body, "body", "b", "", "The value for the variable (reads from standard input if not specified)")
+
+	return cmd
+}
+
+func createRun(opts *shared.PostPatchOptions) error {
+
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("could not create http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+	orgName := opts.OrgName
+	envName := opts.EnvName
+
+	var host string
+	var baseRepo ghrepo.Interface
+	if orgName == "" {
+		baseRepo, err = opts.BaseRepo()
+		if err != nil {
+			return err
+		}
+		host = baseRepo.RepoHost()
+	} else {
+		cfg, err := opts.Config()
+		if err != nil {
+			return err
+		}
+		host, _ = cfg.DefaultHost()
+	}
+	variables, err := shared.GetVariablesFromOptions(opts, client, host, false)
+	if err != nil {
+		return err
+	}
+
+	variableEntity, err := shared.GetVariableEntity(orgName, envName)
+	if err != nil {
+		return err
+	}
+
+	variableApp := shared.App(shared.Actions)
+	if err != nil || variableApp == shared.Unknown {
+		return err
+	}
+
+	if !shared.IsSupportedVariableEntity(variableApp, variableEntity) {
+		return fmt.Errorf("%s variables are not supported for %s", variableEntity, variableApp)
+	}
+
+	repoIdRes := shared.GetRepoIds(client, host, opts.OrgName, opts.RepositoryNames)
+	if repoIdRes.Err != nil {
+		return repoIdRes.Err
+	}
+
+	createc := make(chan createResult)
+	for variableKey, variable := range variables {
+		go func(variableKey string, variable shared.VariablePayload) {
+			var repoIds []int64
+			var visibility string
+			visibility = opts.Visibility
+			//cmd line opt overrides file
+			if len(repoIdRes.Ids) > 0 {
+				repoIds = repoIdRes.Ids
+			} else if variable.Visibility != "" {
+				visibility = variable.Visibility
+				repoIds = variable.Repositories
+			}
+			createc <- createVariable(opts, host, client, baseRepo, variableKey, variable.Value, visibility, repoIds, variableApp, variableEntity)
+		}(variableKey, variable)
+	}
+
+	err = nil
+	cs := opts.IO.ColorScheme()
+	for i := 0; i < len(variables); i++ {
```

Comment:
Again seems like this is not necessary as we only support a single variable.

Reply:
same